### PR TITLE
Fix Bart type hints

### PR DESCRIPTION
**Pull request by gchhablani**
# What does this PR do?

This PR adds type-hints to PLBart PyTorch.

A few queries:
1. What about the configuration files? Do they not need type-hinting?
2. I understand someone is working on the BART model: https://github.com/huggingface/transformers/pull/16270/files. There are some copies on there, can I wait for that PR to be merged and see if there are any fixes needed after doing `make fixup` from their PR?

@Rocketknight1

### CODE REVIEW

**Comment by Rocketknight1** [comment]
But also, thanks for doing this - your PR looks very solid otherwise!

**Comment by gchhablani** [comment]
@Rocketknight1 I have tried to make things a bit more consistent with Bart.
For example:
1. One place was using `torch.FloatTensor` for `encoder_attention_mask`, which should be `torch.LongTensor`.
2. Another discrepancy was that `encoder_hidden_states` were `FloatTensor` in one place, while `Tensor` in another.
3. `input_ids` are actually optional because we can also pass `input_embeds` but it was not present.

PLBart has some additional changes, which may be included. Wdyt?

**Comment by Rocketknight1** [comment]
Hi @gchhablani I think this PR is quite difficult to review right now because of all the changes! If possible, could you close this PR, make sure your local repo is updated to the latest changes, then make a new branch and make the plBART PR there? I think if you do that, and only add annotations to the methods that are missing them, then we shouldn't have the problem of multiple affected files.

Thanks, and I'm sorry for the confusion here!

**Comment by gchhablani** [comment]
Hi @Rocketknight1!

Sure!
What about the issues with BART?

I think the changes in BART and PLBart are the only ones that need reviewing. Wdyt?

**Comment by Rocketknight1** [comment]
I think it's fine to focus on plBART for this PR. The issues with BART aren't serious, and if we decide to be consistent with `Tuple` or `List` across the whole repo, we can do that with a simple find-replace. So in your plBART PR, don't worry about making changes that will also require changes to BART - just annotate missing types for methods that don't have any yet, and that's perfect!

File: src/transformers/models/plbart/modeling_plbart.py [hunk 1]
```diff
@@ -1272,21 +1272,21 @@ def set_output_embeddings(self, new_embeddings):
...
         decoder_inputs_embeds=None,
-        labels: Optional[torch.Tensor] =None,
-        use_cache: Optional[bool] =None,
+        labels: Optional[torch.Tensor] = None,
+        use_cache: Optional[bool] = None,
         output_attentions: Optional[bool] = None,
         output_hidden_states: Optional[bool] = None,
-        return_dict: Optional[bool] = None
+        return_dict: Optional[bool] = None,
     ):
```
Comment (Rocketknight1) [inline src/transformers/models/plbart/modeling_plbart.py:1290]:
```suggestion
    ) -> Union[Tuple[torch.Tensor], Seq2SeqLMOutput]:
```

**Review by Rocketknight1 — APPROVED** [review]
This looks good! I made one suggestion (adding the output type), but other than that ping me when you're ready and I'll merge it!

**Comment by gchhablani** [comment]
@Rocketknight1 Can you please review it now?
Thanks!

**Comment by Rocketknight1** [comment]
Looks great to me, thanks for the PR!

**Pull request merged**